tests/installation/boot_into_snapshot.pm

```diff
@@ -23,10 +23,17 @@ use base "opensusebasetest";
 
 sub run {
     my ($self) = @_;
-    assert_screen 'linux-login', 200;
-    select_console 'root-console';
-    # 1)
-    assert_script_run('touch /etc/NOWRITE;test ! -f /etc/NOWRITE');
+    # for scenarios BOOT_INTO_SNAPSHOT_AFTER_UPGRADE we just need to make sure that root-console is there because
+    # the system is already up and running. For ROLL-BACK it shouldn't check that current snapshot is writeable or not.
+    if (get_var('BOOT_INTO_SNAPSHOT_AFTER_UPGRADE')) {
+        select_console 'root-console';
+    }
+    else {
+        assert_screen 'linux-login', 200;
+        select_console 'root-console';
+        # 1)
+        assert_script_run('touch /etc/NOWRITE;test ! -f /etc/NOWRITE');
```

Comment:
Annoyingly, the test module `boot_into_snapshot` doesn't boot into a snapshot. It is only scheduled in test suite `boot_to_snapshot` and directly after `grub_test`, which boots into a RO snapshot if setting `BOOT_TO_SNAPSHOT` (See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/installation/grub_test.pm#L80)
Then `boot_into_snapshot` expects the SUT to be already booted into a RO snapshot and perform checks and the rollback. (In my opinion, the test module `boot_into_snapshot` should be renamed to `rollback_from_ro_snapshot` to properly describe what it does.

Because `boot_into_snapshot` expects the system booted into a RO snapshot, the check `assert_script_run('touch /etc/NOWRITE;test ! -f /etc/NOWRITE');` is **mandatory**.
If the check fails, it means that the SUT is on a writable snapshot, and that could mean that there is a product bug or a test bug. In this case, _the failure is a test bug_, since the verification run 10.162.23.47/tests/8183, shows the module `boot_into_snapshot` schedule after:
- `grub_test` (which is booting into default writable snapshot)
- `first_boot`
- `opensuse_welcome`

So, the changes on file `tests/installation/boot_into_snapshot.pm` are wrong.